fnet/network.go

```diff
@@ -80,6 +82,8 @@ func ChangeMaxPayloadSize(newMaxPayloadSize int) {
 	if err != nil {
 		log.Errf("Error changing payload size, read for %d random payload failed: %v", newMaxPayloadSize, err)
 	}
+	// slower but we seem to only do this on init anyway, and this guarantees UTF-8
+	Payload = []byte(base64.StdEncoding.EncodeToString(Payload)[:len(Payload)])
```

Comment:
it's not quite same pseudo random, it's now pseudo random utf8, except it can still end up truncated and invalid at the end; what breaks when left as bytes?

Reply:
gRPC will refuse to send it if it's not valid utf-8

Reply:
There must be a type that is bytes, not string?

Reply:
Do you mean here in the generation part, or sending a byte-array directly in the gRPC payload rather than string (changing the protos)? 

For the generation part I didn't see anything buillt-in that could do this. If we generate `< len(payload)` bytes we can manually zero-pad the start of the byte array to be utf-8 compliant.

Reply:
3 different issues

- I don't quite believe there isn't a way to send bytes; maybe if this is for the PingServer proto; we can just change the proto if there is a type mismatch

- Can you show me in the unchanged code how to reproduce the failure? (or give details; refuse to send how exactly?)

- If we really wanted to send utf string then we should indeed generate that rather than truncate (but that's last resort and shouldn't be necessary)

Reply:
> I don't quite believe there isn't a way to send bytes; maybe if this is for the PingServer proto; we can just change the proto if there is a type mismatch

We could totally change the proto to send bytes. I was just trying to avoid changing the protos originally. 

> Can you show me in the unchanged code how to reproduce the failure? (or give details; refuse to send how exactly?) 

I'll get repro steps later today. 

> If we really wanted to send utf string then we should indeed generate that rather than truncate (but that's last resort and shouldn't be necessary)

Agreed – but I think the goal is "create a payload of size n" which `byte[]` is better suited for.